hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -732,31 +732,31 @@ public void setupWriteClient() throws IOException {
     if ((null != schemaProvider)) {
       Schema sourceSchema = schemaProvider.getSourceSchema();
       Schema targetSchema = schemaProvider.getTargetSchema();
-      reInitWriteClient(sourceSchema, targetSchema);
+      createNewWriteClientWithSchema(sourceSchema, targetSchema);
     }
   }
 
-  private void reInitWriteClient(Schema sourceSchema, Schema targetSchema) throws IOException {
+  private void createNewWriteClientWithSchema(Schema sourceSchema, Schema targetSchema) throws IOException {
     LOG.info("Setting up new Hoodie Write Client");
     if (isDropPartitionColumns()) {
       targetSchema = HoodieAvroUtils.removeFields(targetSchema, getPartitionColumns(keyGenerator, props));
     }
     registerAvroSchemas(sourceSchema, targetSchema);
-    HoodieWriteConfig hoodieCfg = getHoodieClientConfig(targetSchema);
-    if (hoodieCfg.isEmbeddedTimelineServerEnabled()) {
+    HoodieWriteConfig writeConfig = getHoodieClientConfig(targetSchema);
+    if (writeConfig.isEmbeddedTimelineServerEnabled()) {
       if (!embeddedTimelineService.isPresent()) {
-        embeddedTimelineService = EmbeddedTimelineServerHelper.createEmbeddedTimelineService(new HoodieSparkEngineContext(jssc), hoodieCfg);
+        embeddedTimelineService = EmbeddedTimelineServerHelper.createEmbeddedTimelineService(new HoodieSparkEngineContext(jssc), writeConfig);
```

Comment:
We should keep one long-running timeline service in DeltaSync, instead of shutting down (by closing the write client) and restarting the timeline server.

In the write client, we should change the behavior of shutting down the timeline server.  If the timeline server instance is passed into the constructor externally (`public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig writeConfig, Option<EmbeddedTimelineService> timelineService)`), the write client should not shut it down, and should let the caller to handle the lifecycle.

Reply:
+1